src/python/pants/pantsd/watchman.py

```diff
@@ -137,7 +142,12 @@ def watch_project(self, path):
 
     :param string path: the path to the watchman project root/pants build root.
     """
-    return self.client.query('watch-project', os.path.realpath(path))
+    # TODO(kwlzn): Add a client.query(timeout=X) param to the upstream pywatchman project.
```

Comment:
can watch_project be called more than once? Would it make sense to reset the timeout to the startup timeout for each call into this?